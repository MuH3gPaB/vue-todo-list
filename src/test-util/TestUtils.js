import {jest} from "@jest/globals";

export const mockModule = function (module, childModules) {
    let mockedModule = {actions: {}, getters: {}, state: {}, modules: childModules};
    Object.keys(module.actions || []).forEach(key => mockedModule.actions[key] = jest.fn());
    Object.keys(module.getters || []).forEach(key => mockedModule.getters[key] = jest.fn());
    Object.keys(module.state || []).forEach(key => mockedModule.state[key] = module.state[key]);
    return mockedModule;
}
