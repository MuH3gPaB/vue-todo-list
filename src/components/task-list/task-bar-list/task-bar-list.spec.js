import {beforeEach, describe, it} from "@jest/globals";
import TaskBarList from './task-bar-list';
import ToDoListModule from '@/store/to-do-list';
import {createLocalVue, mount} from "vue-test-utils";
import Vuex from 'vuex';
import {mockModule} from "@/test-util/TestUtils";
import TaskModule from "@/store/task";

function mockStore(tasks) {
    let tasksModules = tasks.reduce((result, task, idx) => {
        return {
            ...result,
            ['Task' + idx]: mockModule(TaskModule(task))
        }
    }, {});

    return mockModule({}, {
        ToDoListModule: mockModule(ToDoListModule, tasksModules)
    });
}

describe('TaskBarList', () => {
    let localVue;
    let store;
    let wrapper;

    beforeEach(() => {
        localVue = createLocalVue();
        localVue.use(Vuex);
        store = new Vuex.Store(mockStore(testTasks()));

        wrapper = mount(TaskBarList, {
            localVue,
            store,
            propsData: {
                listName: 'testList',
                tasksModulesIds: ['Task0', 'Task1', 'Task2'],
                taskSortOptions: {
                    sortOne: {
                        label: 'SortOne'
                    },
                    sortTwo: {
                        label: 'SortTwo'
                    }
                },
                currentSortCode: 'sortOne'
            }
        });
    });

    it('should match snapshot', () => {
        expect(wrapper.vm.$el).toMatchSnapshot();
    });
});

function testTasks() {
    return [
        {
            id: 0,
            text: 'Test Task1',
            isDone: true,
            createdDate: '2020-10-15T19:24:52.829Z',
            dueDate: '2020-10-15T19:54:52.829Z'
        },
        {
            id: 1,
            text: 'Test Task2',
            isDone: true,
            createdDate: '2020-10-15T19:24:52.829Z',
            dueDate: '2020-10-15T19:54:52.829Z'
        },
        {
            id: 2,
            text: 'Test Task3',
            isDone: true,
            createdDate: '2020-10-15T19:24:52.829Z',
            dueDate: '2020-10-15T19:54:52.829Z'
        },
    ];
}
