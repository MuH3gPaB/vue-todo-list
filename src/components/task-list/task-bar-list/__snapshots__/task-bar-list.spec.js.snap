// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskBarList should match snapshot 1`] = `
<div
  class="task-bar-list"
>
  <div
    class="bar-list-header"
  >
    <p
      class="bar-list-label"
    >
      testList
    </p>
     
    <select
      class="form-control input-lg sort-select"
    >
      <option
        value="sortOne"
      >
        
        SortOne
      
      </option>
      <option
        value="sortTwo"
      >
        
        SortTwo
      
      </option>
    </select>
  </div>
   
  <div
    class="bar-list-body"
  >
    <div
      class="task-bar"
    >
      <div
        class="task"
      >
        <input
          class="form-control checkbox"
          type="checkbox"
        />
         
        <p
          class="text"
        >
          
      Test Task1
    
        </p>
         
        <textarea
          class="text text-input"
          style="display: none;"
        />
         
        <div
          class="time"
        >
          <p
            class="created"
          >
            10:24 PM
          </p>
           
          <p
            class="completed"
          >
            10:54 PM
          </p>
        </div>
      </div>
       
      <div
        class="delete"
        style="display: none;"
      />
    </div>
    <div
      class="task-bar"
    >
      <div
        class="task"
      >
        <input
          class="form-control checkbox"
          type="checkbox"
        />
         
        <p
          class="text"
        >
          
      Test Task2
    
        </p>
         
        <textarea
          class="text text-input"
          style="display: none;"
        />
         
        <div
          class="time"
        >
          <p
            class="created"
          >
            10:24 PM
          </p>
           
          <p
            class="completed"
          >
            10:54 PM
          </p>
        </div>
      </div>
       
      <div
        class="delete"
        style="display: none;"
      />
    </div>
    <div
      class="task-bar"
    >
      <div
        class="task"
      >
        <input
          class="form-control checkbox"
          type="checkbox"
        />
         
        <p
          class="text"
        >
          
      Test Task3
    
        </p>
         
        <textarea
          class="text text-input"
          style="display: none;"
        />
         
        <div
          class="time"
        >
          <p
            class="created"
          >
            10:24 PM
          </p>
           
          <p
            class="completed"
          >
            10:54 PM
          </p>
        </div>
      </div>
       
      <div
        class="delete"
        style="display: none;"
      />
    </div>
  </div>
   
  <div
    class="bar-list-footer"
  >
    <a
      class="bar-list-clear"
      href="#"
    >
      
      Clear "testList" list
    
    </a>
  </div>
</div>
`;
